### ISSUE TITLE
Offset/padding?

### ISSUE THREAD

**Issue by devjta** [peterLaurence/MapView#44]
Hey!

Turns out, for whatever reasons our tiles have 1 pixel offset on each side, resulting in a 258x258 tile.

Would there be a way to add such offsets? If you dont want to add features in the non-compose version, I fully understand (we will probably switch to it at the end off the year), but could I even do such a thing, without redrawing the bitmaps?

I already need to fork your version anyway, because of the tileSize being the widthXheight for the bitmap, which didnt work.

I found in the mapConfiguration a setPadding, but didnt had any effect.

Problem is, that on high zoom ratio, the images/tiles are not stiched correctly.

BR

**Comment by p-lr** [comment]
Hi, I'm not sure I understand correctly. Are your tiles 258x258px?
If this is the case, setting the `tileSize` of `MapViewConfiguration` to 258 should work.
Is there something I missed?

**Comment by p-lr** [comment]
Maybe you thought that the `tileSize` parameter was number of pixels. Actually, this is the width (or height) in pixels of tiles (as MapView only supports square tiles).

**Comment by p-lr** [comment]
Besides that, you're right about the compose vs view library. I'm only fixing issues on MapView and adding features to MapCompose.

**Comment by devjta** [comment]
We have non rectanlge tiles (the one at the edges). We fixed it, by drawing them on a white background.

The problem with setting it to 258, is probably than the calculation of the pins. As the webapp (openlayers) and iOS are using tileSize = 256, with the 258x258 tiles (iOS also has that bug, when zooming much. Only OpenLayers handle it correctly).

Whats even more strange: If I set the tileSize to 258, the issue is still present. On full zoom, its not stitched correctly.

Its really strange.. need to investigate it..

What fixes it, but is massive performance loss would be, adding that in the TileCollector:
`            val tile = Tile(spec.zoom, spec.row, spec.col, spec.subSample).apply {
                //if height or width is not 256, make the bitmap like that
                if(bitmap.height != 256 || bitmap.width != 256){
                    this.bitmap = Bitmap.createBitmap(256, 256, Bitmap.Config.ARGB_8888)
                    val canvas = Canvas(this.bitmap)
                    canvas.drawColor(Color.WHITE)
                    canvas.drawBitmap(bitmap, 0.0f, 0.0f, null)
                }else {
                    this.bitmap = bitmap
                }
            }
`

This would also fix the issue, with tiles not being squares. But as I wrote, i basically redraw each bitmap and create a bitmap for each tile..

**Comment by p-lr** [comment]
FWIW, creating bitmaps of size 258x258 with `Bitmap.createBitmap(258, 258, Bitmap.Config.ARGB_8888)` may work for 256x256 tiles. However, I have to admit, MapView was never designed to display tiles of different sizes.

**Comment by devjta** [comment]
@p-lr Hey, just wanted to let you know, you basically had already everything which I needed.

In the TileCanvasView.kt in the drawTiles function you had this line:

`canvas.drawBitmap(tile.bitmap, null, dest, paint)`

I looked then at the API documentation and basically I can also provide a Rect for the source. So fix was that:

`private val source = Rect(1,1,257,257)`

and in drawTiles then

`canvas.drawBitmap(tile.bitmap, source, dest, paint)`


I added a detection logic, to see if width or height is 258.
